### ISSUE TITLE
Please bring back the custom tiltebar!

### ISSUE THREAD

**Issue by Minibus93** [amanharwara/altus#125]
Hi, first of all please tag this as "enhancement" or any other tag that isn't a bug cause it is not.

I'd like to ask to add the option to use the custom titlebar in 4.0.0 since the normal titlebar (which is white in my case) is not very nice looking and doesn't fit the theme (to me atleast) if using dark plus theme.

I really liked the custom titlebar in 3.24 and it was a dealbreaker to discover that in 4.0.0 it wasn't there. (I went back to 3.24 for this)

Has this feature any chance to be supported again? Maybe just like the option that 3.24 has, it lets you choose if you want to use or not the custom titlebar.

Adding a screenshot to clarify what is titlebar in my mind cause I'm pretty sure that titlebar is the correct name but one never knows 
![altus titlebar I guess](https://user-images.githubusercontent.com/73554423/107054938-04208d00-67d1-11eb-8c82-ffb707f68021.png)

**Comment by amanharwara** [comment]
Is this really that big of a dealbreaker? I removed it because it's an additional dependency from which I don't see a real benefit.

**Comment by Minibus93** [comment]
For me, yes it is. 

Default titlebar doesn't fit the theme and even tho functionality-wise app still goes wonderful, the custom titlebar gave that little bonus touch especially on windows 10, where consistency is only a very far dream. Since Altus is an app that I mostly have 24/7 open because I use it very often it's kinda out of place to see my default title bar.

If handling that dependency isn't too much of a pita, I'm sure there would be a lot of folks that would appreciate its return

**Comment by amanharwara** [comment]
I see. One more reason I dropped the dependency was that it requires me to enable the `remote` module in Electron which has been deprecated and will be removed in upcoming Electron versions. I'll try & see if I can make the custom title bar myself, otherwise you'll have to wait & see if the dependency maintainer updates it to not use that module any more.

**Comment by Minibus93** [comment]
Thanks for the effort! <3

**Comment by Melvin-Abraham** [comment]
> I see. One more reason I dropped the dependency was that it requires me to enable the `remote` module in Electron which has been deprecated and will be removed in upcoming Electron versions. I'll try & see if I can make the custom title bar myself, otherwise you'll have to wait & see if the dependency maintainer updates it to not use that module any more.

Keeping in view the deprecation of `remote`, a safe and better alternative to that would be `ipcMain` and `ipcRenderer` and I am sure that custom titlebar can be implemented with the help of IPC.

**Comment by amanharwara** [comment]
>
> Keeping in view the deprecation of `remote`, a safe and better alternative to that would be `ipcMain` and `ipcRenderer` and I am sure that an in-house custom titlebar can be implemented with the help of IPC without any dependency :)

Yeah, IPC is probably the way to go for implementing this. Right now, I'm working on some other projects, so I don't have time to implement this. If anybody else is willing to implement it, I'll be more than happy to look at the PR & accept it. Otherwise, I'll just try to implement it when I get the time.

**Comment by amanharwara** [comment]
I've created the https://github.com/amanharwara/altus/tree/custom-titlebar branch for work on this. So far, I've managed to create the whole title bar but without the menu which means it's kind of pointless right now. Right now I can't seem to find a way to implement the menu using IPC without remote. If anyone else can do it, please create a PR.

**Comment by amanharwara** [comment]
[v4.2.0](https://github.com/amanharwara/altus/releases/tag/4.2.0) adds back the Custom Titlebar setting. It is turned off by default.

Do note that this implementation of the titlebar isn't feature-complete yet but the basic functionality works.

**Comment by Minibus93** [comment]
I'm gonna install it as soon as I'm back home and test it! 

Thank you for all the efforts!
